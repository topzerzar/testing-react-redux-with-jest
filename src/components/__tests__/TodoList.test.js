import React from 'react'
import { shallow } from 'enzyme'
import TodoList from '../TodoList'

// Describe what to test
describe('TodoItem', () => {
  let component
  let props
  beforeEach(() => {
    props = [
        { id: 1, text: 'feed cat', complete: false },
        { id: 1, text: 'feed cat', complete: false },
    ]
    component = shallow(<TodoList {...props} />)
  })
  // Test Scenario
  it('matches its snapshot', () => {
    expect(component).toMatchSnapshot()
  })

  it('render correct structure', () => {
    expect(component.is('ul')).toBe(true)
    expect(component.hasClass('todo-list')).toBe(true)
  })

  it('renders TodoItem from props', () => {
    expect(component.find('TodoItem').length).toBe(props.todos.length)
  })
})

