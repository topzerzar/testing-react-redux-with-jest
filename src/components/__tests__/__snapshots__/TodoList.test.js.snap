// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem matches its snapshot 1`] = `undefined`;
